include/ap/argument_parser.hpp

```diff
@@ -67,32 +115,100 @@ struct argument_name {
         return os;
     }
 
+    /// @brief The long name of the argument.
     const std::string_view name;
+
+    /// @brief The optional short name of the argument.
     const std::optional<std::string_view> short_name;
 };
 
 
 
+/**
+ * @brief Interface for argument configuration.
+ */
 class argument_interface {
 public:
+    /**
+     * @brief Sets the help message for the argument.
+     * @param help Help message.
+     * @return Reference to the argument_interface.
+     */
     virtual argument_interface& help(std::string_view) = 0;
+
+    /**
+     * @brief Sets whether the argument is required.
+     * @param required True if the argument is required.
+     * @return Reference to the argument_interface.
+     */
     virtual argument_interface& required(bool) = 0;
+
+    /**
+     * @brief Sets the default value for the argument.
+     * @param value Default value.
+     * @return Reference to the argument_interface.
+     */
     virtual argument_interface& default_value(const std::any&) = 0;
 
+    /// @brief Argument interface destructor.
     virtual ~argument_interface() = default;
 
     friend class ::ap::argument_parser;
 
 protected:
+    /**
+     * @brief Sets the value for the argument.
+     * @param value The value to set.
+     * @return Reference to the argument_interface.
+     */
     virtual argument_interface& value(const std::any&) = 0;
 
+    /**
+     * @brief Checks if the argument is positional.
```

Comment:
In getter functions you can omit the `@brief` description. (Use only `@return` - DRY)